server/lib/nl/detection/utils.py

```diff
@@ -309,3 +309,12 @@ def find_word_boundary(haystack: str, needle: str):
   if match:
     return match.start()
   return -1
+
+
+# Replaces strings in a query given a dictionary where key is the original
+# string and value is the replacement string to use
+def replace_strings_in_query(query: str, replacements: Dict[str, str]) -> str:
+  processed_query = query
+  for orig, new in replacements.items():
+    processed_query = re.sub(rf"\b{orig}\b", new, processed_query)
```

Comment:
nit: comment - `\b` is for ensuring replacements happen in word boundary ?

Reply:
done